src/Spice86/Program.cs

```diff
@@ -34,37 +30,9 @@ public static void RunWithOverrides<T>(string[] args, string expectedChecksum) w
     /// </summary>
     /// <param name="args">The command-line arguments.</param>
     [STAThread]
-    public static void Main(string[] args)
-    {
-        Configuration configuration = CommandLineParser.ParseCommandLine(args);
-
-        if (!configuration.HeadlessMode)
-        {
-            OxyPlotModule.EnsureLoaded();
-            BuildAvaloniaApp().StartWithClassicDesktopLifetime(args, Avalonia.Controls.ShutdownMode.OnMainWindowClose);
-        }
-        else
-        {
-            ServiceProvider serviceProvider = Startup.StartupInjectedServices(args);
-            ILoggerService? loggerService = serviceProvider.GetService<ILoggerService>();
-            if (loggerService is null)
-            {
-                throw new InvalidOperationException("Could not get logging service from DI !");
-            }
-
-            ProgramExecutor programExecutor = new ProgramExecutor(loggerService, null, configuration);
-            programExecutor.Run();
-        }
-    }
-
-    /// <summary>
-    /// Configures and builds an Avalonia application instance.
-    /// </summary>
-    /// <returns>The built <see cref="AppBuilder"/> instance.</returns>
-    public static AppBuilder BuildAvaloniaApp()
-    {
-        return AppBuilder.Configure<App>()
-            .UsePlatformDetect()
-            .LogToTrace();
+    public static void Main(string[] args) {
+        using TopLevelContainer topLevelContainer = new();
```

Comment:
If we resolve the configuration first, we can inject it into the container and have it available everywhere.
See https://github.com/YairHalberstadt/stronginject#providing-registrations-at-runtime-or-integrating-with-other-ioc-containers

And if we make the Configuration an `IConfiguration`  we can use commandline, jsonfile, environment variable, etc. `IConfigurationBuilder`s

Reply:
I was about to use IConfiguration, but feared if was too much of a change at once.

I don't know how well it will mix the [Option] tags from Configuration.cs, which is exposed for command line parsing and to provide commandeline help.

Reply:
Yes doing IConfiguration or full-on Options pattern heree and now would not make sense.
But moving the Configuration creation to _before_ the container initialization does feel like something that belongs in this PR.

Reply:
I moved the Configuration creation before the line that created the Container.
I pass it the configuration on initialization, and use its Configuration property to call the Startup.StartApp.

But I've yet to use that property everywhere in the UI layer, as it is only available in Program.cs since topLevelContainer is a local variable.
To avoid a warning in Program.cs saying "captured variable disposed in the outer scope" I don't dispose of the container in Program.cs anymore...